### PR TITLE
Decay for return value

### PR DESCRIPTION
**Pull request by oskmeister**

### CODE REVIEW

File: test/src/test_eq.cpp [hunk 1]
```diff
@@ -64,6 +65,11 @@ BOOST_AUTO_TEST_CASE(json_codec_eq_should_construct_with_helper) {
 /*
  * Decoding
  */
+BOOST_AUTO_TEST_CASE(json_codec_eq_should_decode_local_variable) {
+  int value = 5;
+  const auto codec = eq(value);
+  BOOST_CHECK_EQUAL(test_decode(codec, "5"), value);
```
Comment (pereckerdal) [inline test/src/test_eq.cpp]: does this test fail without the code change?

File: test/src/test_eq.cpp [hunk 2]
```diff
@@ -64,6 +65,11 @@ BOOST_AUTO_TEST_CASE(json_codec_eq_should_construct_with_helper) {
 /*
  * Decoding
  */
+BOOST_AUTO_TEST_CASE(json_codec_eq_should_decode_local_variable) {
```
Comment (punchfox) [inline test/src/test_eq.cpp]: Missing newline.

Comment (oskmeister) [inline test/src/test_eq.cpp]: 🕴

File: test/src/test_eq.cpp (hunk 1 above)
Comment (oskmeister) [inline test/src/test_eq.cpp]: It doesn't even compile.

**Pull request merged**